### PR TITLE
Pxb 8.0.21.merge

### PR DESCRIPTION
**Pull request by rahulmalik87**

### CODE REVIEW

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtrabackup?pullRequest=914) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtrabackup?pullRequest=914) before we can accept your contribution.<br/>**3** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: Sudokamikaze<br/>:white_check_mark: patrickbirch<br/>:white_check_mark: rahulmalik87<br/>:x: blaudden<br/>:x: Mayank Prasad<br/>:x: arnabray21<br/>:x: lkshminarayanan<br/>:x: jensevenb<br/>:x: kwedde<br/>:x: bjornmu<br/>:x: ltangvald<br/>:x: dahlerlend<br/>:x: bkandasa<br/><hr/>**Mayank Prasad** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtrabackup?pullRequest=914) it.</sub>

File: storage/innobase/xtrabackup/test/t/all_files.sh [hunk 1]
```diff
@@ -29,12 +29,6 @@ diff -u <( ( ( cd $dir1; find . | grep -Ev $ign_list )
 ./xtrabackup_tablespaces
 EOF
 
-if is_xtradb ; then
-    XTRA_DOUBLEWRITE="./xb_doublewrite"
-else
-    XTRA_DOUBLEWRITE=""
-fi
-
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/test/t/all_files.sh]: we need this isn't it? if PXB is run against PS 8.0.19 or lower ?

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

**Comment by satya-bodapati** [comment]
@rahulmalik87 Also, please follow different merge style (similar to PS) i.e the first merge commit should commit with all conflict markers as is. Next commits should say "fixing conflicts" and then fixing tests..

File: storage/innobase/xtrabackup/test/t/all_files.sh (hunk 1 above)
Comment (rahulmalik87) [reply storage/innobase/xtrabackup/test/t/all_files.sh]: I have bumped the testing version to PS 8.0.20.

Comment (satya-bodapati) [reply storage/innobase/xtrabackup/test/t/all_files.sh]: but I think the test case can be changed to handle both. How else would we test PXB against lower versions and its compatibility?

**Comment by satya-bodapati** [comment]
Can you please get Jenkins run for 8.0.19 or lower (v3), 8.0.20 (v4) and 8.0.21 (current)?

**Comment by rahulmalik87** [comment]
https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param-medium/lastCompletedBuild/testReport/
ps 8.0.20 and ms 8.0.20

**Review by satya-bodapati — CHANGES_REQUESTED** [review]
waiting for Jenkins links

**Comment by satya-bodapati** [comment]
> https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param-medium/lastCompletedBuild/testReport/
> ps 8.0.20 and ms 8.0.20

test failures that should be scrutinized: 

t/bug1343722.sh
t/all_files.sh
t/binlog_info.sh
t/ib_databases.sh

@rahulmalik87 ^

**Comment by rahulmalik87** [comment]
MySQL 8.0.21,  PS 8.0.18 https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param/139/#showFailuresLink

 MySQL 8.0.20,  PS 8.0.20 https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param/136/#showFailuresLink

MySQL 8.0.18 , PS 8.0.18 https://pxb.cd.percona.com/view/PXB%208.0/job/percona-xtrabackup-8.0-param/137/#showFailuresLink

**Review by satya-bodapati — APPROVED** [review]
Looks good. Only one test failure (pxb-1824.sh) which is claimed to be random.

Also suggest to do the future merges to do PS way
1. Merge from upstream and commit with conflict markers
2. One commit to resolve conflicts
3. Additional code fixes, test fixes

**Pull request merged**